### PR TITLE
Added repositories for deb/rpm packages on buildkite

### PR DESCRIPTION
**Pull request by stone-w4tch3r**
In issue #298 we discussed distributing rpm/deb packages via dedicated repositories. 

Initial idea was to use packagecloud, but since it is now not accessible for new users, buildkite was used as a replacement. Actually, it was quite hard to handle all it's bugs, but now it works!

You can try test packages here:  [rpm](https://buildkite.com/organizations/stone-w4tch3r/packages/registries/sourcegit-fork-test-rpm) and [deb](https://buildkite.com/organizations/stone-w4tch3r/packages/registries/sourcegit-fork-test-deb).

To make it all work, you need: 
- register an organization named **sourcegit** on buildkite
- create two repositories: **sourcegit-deb** and **sorcegit-rpm**
- in personal settings create a token with **write_packages** and **read_packages** scopes
- configure **BUILDKITE_TOKEN** secret in repository settings
- push a release!

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -24,6 +24,14 @@ jobs:
     uses: ./.github/workflows/package.yml
     with:
       version: ${{ needs.version.outputs.version }}
+  publish-packages:
+    needs: [package, version]
+    name: Publish Packages
+    uses: ./.github/workflows/publish-packages.yml
+    with:
+      version: ${{ needs.version.outputs.version }}
```
Comment (aikawayataro) [inline .github/workflows/release.yml:32]: Not used by `publish-packages.yml`

**Comment by aikawayataro** [comment]
Looks like a working solution for #298

**Comment by aikawayataro** [comment]
BTW I couldn't find a workflow run for this in your fork. Can you do a quick release test there?

File: .github/workflows/release.yml (hunk 1 above)
Comment (stone-w4tch3r) [reply .github/workflows/release.yml]: removed

File: .github/workflows/publish-packages.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: Publish to Buildkite
+on:
+ workflow_call:
+   inputs:
+     version:
```
[the remark this answers is not in the document]
Comment (stone-w4tch3r) [reply .github/workflows/publish-packages.yml]: removed

**Comment by stone-w4tch3r** [comment]
> BTW I couldn't find a workflow run for this in your fork. Can you do a quick release test there?

It... disappeared somehow? Anyway look at new release: https://github.com/stone-w4tch3r/sourcegit-fork/actions/runs/12139933496

**Comment by stone-w4tch3r** [comment]
Note that actual push failed because I already changed repository/organization names in an action. But previous packages were pushed successfully

**Comment by aikawayataro** [comment]
I see, but action is not failed because error is ignored. Would you add `--fail` to curl and see if workflow fails properly with it :grin:

**Comment by stone-w4tch3r** [comment]
Now proper error
https://github.com/stone-w4tch3r/sourcegit-fork/actions/runs/12140279854

**Comment by aikawayataro** [comment]
Looks good

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -24,6 +24,12 @@ jobs:
     uses: ./.github/workflows/package.yml
     with:
       version: ${{ needs.version.outputs.version }}
+  publish-packages:
+    needs: [package, version]
```
[the remark this answers is not in the document]
Comment (stone-w4tch3r) [reply .github/workflows/release.yml]: fixed

**Comment by love-linger** [comment]
Hi everyone, the new CI workflow `release` run failed. Please help me fix it.

**Comment by sourcegit-scm** [comment]
I've pushed these packages manually, and modify the `release.yml` with following changes

```diff
@@ -24,8 +24,9 @@ jobs:
     uses: ./.github/workflows/package.yml
     with:
       version: ${{ needs.version.outputs.version }}
   publish-packages:
+    needs: [package, version]
     name: Publish Packages
     uses: ./.github/workflows/publish-packages.yml
     secrets:
       BUILDKITE_TOKEN: ${{ secrets.BUILDKITE_TOKEN }}
```

Is it right?

**Comment by stone-w4tch3r** [comment]
> I've pushed these packages manually, and modify the `release.yml` with following changes
> 
> ```diff
> @@ -24,8 +24,9 @@ jobs:
>      uses: ./.github/workflows/package.yml
>      with:
>        version: ${{ needs.version.outputs.version }}
>    publish-packages:
> +    needs: [package, version]
>      name: Publish Packages
>      uses: ./.github/workflows/publish-packages.yml
>      secrets:
>        BUILDKITE_TOKEN: ${{ secrets.BUILDKITE_TOKEN }}
> ```
> 
> Is it right?

Yep. Checked, without this line publish fails. You need to return it back to release.yml

**Comment by love-linger** [comment]
the original PR missed this line. I've added it to `develop` branch

**Comment by stone-w4tch3r** [comment]
I can't find repos on buildkite. What links to use? Checked `https://buildkite.com/organizations/sourcegit/packages/registries/sourcegit-deb`, but only prompt for login is present

**Comment by love-linger** [comment]
> https://buildkite.com/organizations/sourcegit/packages/registries/sourcegit-deb

try it again?

**Comment by stone-w4tch3r** [comment]
Added appimagehub into readme #798

**Pull request merged**